Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -493,17 +493,17 @@ inline bool ShouldDraw() {
         return SUCCEEDED(_firstErrorHr) && CurrentGState().ShouldDraw();
     }
 
-    inline void PushBeginDraw() {
-        if ((_beginEndDrawDepth)++ == 0) {
+    inline void PushBeginDraw(bool forceBeginDraw = false) {
+        if ((_beginEndDrawDepth)++ == 0 || forceBeginDraw) {
```

Comment:
If we're going to go with the "override" flag, putting it _first_ will save unnecessary manipulation of the counter.